aiohttp/web_urldispatcher.py

```diff
@@ -353,23 +354,18 @@ def register_route(self, route: "ResourceRoute") -> None:
         assert isinstance(
             route, ResourceRoute
         ), f"Instance of Route class is required, got {route!r}"
-        self._routes.append(route)
+        if route.method == hdrs.METH_ANY:
+            self._any_route = route
+        else:
+            self._allowed_methods.add(route.method)
```

Comment:
Why not use `self._routes.keys()`?

Reply:
To avoid making a new set every time since the api currently wants a set returned. 

Maybe could be refactored more to change the type to allowed to be a Collection but I wanted to keep the change compatible as possible

Reply:
Technically `dict.keys()` provides `collections.abc.Set`, switching from `set[str]` to `collections.abc.Set[str]` is almost safe.

Reply:
up to you